receiver/kubeletstatsreceiver/internal/kubelet/network.go

```diff
@@ -12,11 +12,21 @@ import (
 
 type getNetworkDataFunc func(s *stats.NetworkStats) (rx *uint64, tx *uint64)
 
-func addNetworkMetrics(mb *metadata.MetricsBuilder, networkMetrics metadata.NetworkMetrics, s *stats.NetworkStats, currentTime pcommon.Timestamp) {
+type getInterfaceDataFunc func(s *stats.InterfaceStats) (rx *uint64, tx *uint64)
+
+func addNetworkMetrics(mb *metadata.MetricsBuilder, networkMetrics metadata.NetworkMetrics, s *stats.NetworkStats, currentTime pcommon.Timestamp, allInterfaces bool) {
 	if s == nil {
 		return
 	}
 
+	if allInterfaces {
+		for i := range s.Interfaces {
+			recordInterfaceDataPoint(mb, networkMetrics.IO, &s.Interfaces[i], getInterfaceIO, currentTime)
+			recordInterfaceDataPoint(mb, networkMetrics.Errors, &s.Interfaces[i], getInterfaceErrors, currentTime)
+		}
+		return
```

Comment:
Shall we add comment note here that the default interface is also included in the interfaces' list, hence we return immediately?

Reply:
Good point, added